flink-python/src/main/java/org/apache/flink/table/runtime/operators/python/aggregate/arrow/batch/BatchArrowPythonGroupAggregateFunctionOperator.java

```diff
@@ -71,6 +71,7 @@ protected void invokeCurrentBatch() throws Exception {
             elementCount += currentBatchCount;
             checkInvokeFinishBundleByCount();
             currentBatchCount = 0;
+            arrowSerializer.reInitializeWriter(baos);
```

Comment:
Is it possible to call `arrowSerializer.reInitializeWriter(baos);` inside `finishCurrentBatch`? Then we could avoid this change.